install/installer/pkg/components/public-api-server/deployment.go

```diff
@@ -18,21 +16,6 @@ import (
 )
 
 func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
-	var experimentalCfg *experimental.Config
-
-	_ = ctx.WithExperimental(func(ucfg *experimental.Config) error {
```

Comment:
This check has been effectively hoisted higher to the Objects function (the main entry point for rendering anything for `public-api`). This ensures that future objects are also protected with the same config.